File: Java_Advanced/gsapi/src/main/java/br/com/fiap/gs/gsapi/dto/request/ContatoRequestDTO.java
package br.com.fiap.gs.gsapi.dto.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public class ContatoRequestDTO {

    @NotBlank(message = "O DDD não pode estar em branco.")
    @Size(min = 2, max = 3, message = "O DDD deve ter entre 2 e 3 caracteres.")
    private String ddd;

    @NotBlank(message = "O telefone não pode estar em branco.")
    @Size(min = 8, max = 15, message = "O telefone deve ter entre 8 e 15 caracteres.")
    @Pattern(regexp = "^[0-9]+$", message = "O telefone deve conter apenas números.")
    private String telefone;

    @NotBlank(message = "O celular não pode estar em branco.")
    @Size(min = 9, max = 15, message = "O celular deve ter entre 9 e 15 caracteres.")
    @Pattern(regexp = "^[0-9]+$", message = "O celular deve conter apenas números.")
    private String celular;

    @NotBlank(message = "O WhatsApp não pode estar em branco.")
    @Size(min = 9, max = 15, message = "O WhatsApp deve ter entre 9 e 15 caracteres.")
    @Pattern(regexp = "^[0-9]+$", message = "O WhatsApp deve conter apenas números.")
    private String whatsapp;

    @NotBlank(message = "O e-mail não pode estar em branco.")
    @Email(message = "O e-mail deve ser válido.")
    @Size(max = 255, message = "O e-mail não pode exceder 255 caracteres.")
    private String email;

    @NotBlank(message = "O tipo de contato não pode estar em branco.")
    @Size(max = 50, message = "O tipo de contato não pode exceder 50 caracteres.")
    private String tipoContato;

    // Getters e Setters
    public String getDdd() { return ddd; }
    public void setDdd(String ddd) { this.ddd = ddd; }
    public String getTelefone() { return telefone; }
    public void setTelefone(String telefone) { this.telefone = telefone; }
    public String getCelular() { return celular; }
    public void setCelular(String celular) { this.celular = celular; }
    public String getWhatsapp() { return whatsapp; }
    public void setWhatsapp(String whatsapp) { this.whatsapp = whatsapp; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getTipoContato() { return tipoContato; }
    public void setTipoContato(String tipoContato) { this.tipoContato = tipoContato; }
}

File: Advanced_Business_Development_with.NET/gsApi/gsApi/dto/request/ContatoRequestDto.cs
// File: DTOs/Request/ContatoRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace gsApi.DTOs.Request
{
    /// <summary>
    /// DTO para criar/atualizar um contato via API.
    /// </summary>
    public class ContatoRequestDto
    {
        [Required(ErrorMessage = "O DDD é obrigatório.")]
        [StringLength(3, MinimumLength = 2, ErrorMessage = "O DDD deve ter entre 2 e 3 dígitos.")]
        public string Ddd { get; set; } = string.Empty;

        [Required(ErrorMessage = "O Telefone é obrigatório.")]
        [StringLength(15, ErrorMessage = "O Telefone não pode exceder 15 caracteres.")]
        public string Telefone { get; set; } = string.Empty;

        [Required(ErrorMessage = "O Celular é obrigatório.")]
        [StringLength(15, ErrorMessage = "O Celular não pode exceder 15 caracteres.")]
        public string Celular { get; set; } = string.Empty;

        [Required(ErrorMessage = "O Whatsapp é obrigatório.")]
        [StringLength(15, ErrorMessage = "O Whatsapp não pode exceder 15 caracteres.")]
        public string Whatsapp { get; set; } = string.Empty;

        [Required(ErrorMessage = "O Email é obrigatório.")]
        [EmailAddress(ErrorMessage = "Formato de email inválido.")]
        public string Email { get; set; } = string.Empty;

        [Required(ErrorMessage = "O Tipo de Contato é obrigatório.")]
        [StringLength(50, ErrorMessage = "O Tipo de Contato não pode exceder 50 caracteres.")]
        public string TipoContato { get; set; } = string.Empty;
    }
}
